docs/usage/writing-logic-thunks.mdx

```diff
@@ -0,0 +1,463 @@
+
+</DetailedExplanation>
+
+### Thunk Use Cases
+
+Because thunks are a general-purpose tool that can contain arbitrary logic, they can be used for a wide variety of purposes. The most common use cases are:
+
+- Moving complex logic out of components
+- Making async requests or other async logic
+- Writing logic that needs to dispatch multiple actions in a row or over time
+- Writing logic that needs access to `getState` to make decisions or include other state values in an action
+
+Thunks are "one-shot" functions, with no sense of a lifecycle. They also cannot see other dispatched actions. So, they should not generally be used for initializing persistent connections like websockets, and you can't use them to respond to other actions.
+
+**Thunks are best used for complex synchronous logic, and simple to moderate async logic such as making a standard AJAX request and dispatching actions based on the request results.**
+
+## Redux Thunk Middleware
+
+Dispatching thunk functions requires that the [`redux-thunk` middleware](https://github.com/redujs/redux-thunk) has been added to the Redux store as part of its configuration.
+
+### Adding the Middleware
+
+The [Redux Toolkit `configureStore` API automatically adds the thunk middleware during store creation](../tutorials/fundamentals/part-8-modern-redux.md#using-configurestore), so it should typically be available with no extra configuration needed.
+
+If you need to add the thunk middleware to a store manually, that can be done by [passing the thunk middleware to `applyMiddleware()`](..//tutorials/fundamentals/part-6-async-logic.md#configuring-the-store) as part of the setup process.
+
+### How Does the Middleware Work?
+
+[Redux middleware are all written as a series of 3 nested functions](../tutorials/fundamentals/part-4-store.md#writing-custom-middleware):
+
+- The outer function receives a "store API" object with `{dispatch, getState}`
+- The middle function receives the `next` middleware in the chain (or the actual `store.dispatch` method)
+- The inner function will be called with each `action` as it's passed through the middleware chain
+
+It's important to note that middleware can be used to allow passing values that are _not_ action objects into `store.dispatch()`, as long as the middleware intercepts those values and does not let them reach the reducers.
```

Comment:
I think what you're saying here is .... "The  thunk middleware is able to handle thunks and not just action objects, because..." But instead you're speaking in a more generic way here and I found it confusing